tests/rest_api/test_rest_api_source_offline.py

```diff
@@ -288,3 +289,32 @@ def test_load_mock_api_typeddict_config(mock_api_server):
 
     assert table_counts["posts"] == 100
     assert table_counts["post_comments"] == 5000
+
+
+def test_posts_with_inremental_date_transformation(mock_api_server) -> None:
```

Comment:
This is great!

Reply:
An even better way might be to spy on the request client and assert that it was called with the transformed query string params. If you agree, we could do that in a refactoring PR and get the feature shipped for now.

Reply:
done in https://github.com/dlt-hub/verified-sources/pull/519